packages/backend/src/invoice/service.ts

```diff
@@ -50,13 +53,18 @@ async function getInvoice(
 
 async function createInvoice(
   deps: ServiceDependencies,
-  userId: string
+  accountId: string,
+  description: string,
+  expiresAt?: Date,
+  trx?: Transaction
 ): Promise<Invoice> {
-  const user = await deps.userService.get(userId)
-  const subAccount = await deps.accountService.createSubAccount(user.accountId)
-  return Invoice.query(deps.knex).insertAndFetch({
-    userId: userId,
-    accountId: subAccount.id,
+  const subAccount = await deps.accountService.createSubAccount(accountId)
```

Comment:
I think this will work fine even though `trx` isn't passed to `accountService.createSubAccount`, but it is a little awkward. (It may create a dangling account if the invoice's transaction fails.) Too bad javascript doesn't have dynamically-scoped variables :stuck_out_tongue:

Reply:
@sentientwaffle I will rather just make the subaccount also be wrapped in the transaction. I just missed it TBH.

> Too bad javascript doesn't have dynamically-scoped variables 
You'll be waiting a long time 😆